### PR TITLE
Email Verification Code

### PR DESCRIPTION
**Pull request by pondpiu**
- Modify email verification flow to use verification code instead of verification link

Note: Need FE change to use this new flow as well, current flow will break if this is merged without FE change

### CODE REVIEW

File: src/services/EmailService.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ export class EmailService implements IEmailService {
         const link = `${this.config.get(
             ConfigSettings.WEBSITE_URL
         )}/verify-email?token=${token}`;
-        this.logger.log(`${email} Verification email sent`);
+        this.logger.log(`${email} Verification link email sent`);
```
Comment (bitcoinbrisbane) [inline src/services/EmailService.ts:29]:
```suggestion
        this.logger.log(`${email} Verification code email sent`);
```

**Review by bitcoinbrisbane — COMMENTED** [review]
One minor thing

**Review by bitcoinbrisbane — APPROVED** [review]

**Review by bitcoinbrisbane — APPROVED** [review]

File: src/user/user.service.ts [hunk 2]
```diff
@@ -73,21 +77,22 @@ export class UserService {
             // Do nothing if user already exists and email is verified
             if (user && user.emailVerified) return;
 
-            const payload = { email: newUser.email };
-            const token = this.jwtService.sign(payload);
+            const sixDigitCode = Math.floor(
```
Comment (brettvniekerk) [inline src/user/user.service.ts:80]: unit test? probs overkill

**Review by brettvniekerk — APPROVED** [review]
LGTM just one thing to consider

**Pull request merged**